tests/models/musicgen/test_modeling_musicgen.py

```diff
@@ -510,6 +512,108 @@ class MusicgenTest(ModelTesterMixin, GenerationTesterMixin, PipelineTesterMixin,
     def setUp(self):
         self.model_tester = MusicgenTester(self)
 
+    # Not to test with `MusicgenForConditionalGeneration` as the model tester doesn't prepare `input_values` and
+    # `padding_mask` (and `torchscript` hates `None`).
+    def _create_and_check_torchscript(self, config, inputs_dict):
```

Comment:
Would it not be easier to simply set `test_torchscript=False` when we init the model tester for `MusicgenTest`:
https://github.com/huggingface/transformers/blob/906afa1d5c6054a641cb6abb009cdec732a5a094/tests/models/musicgen/test_modeling_musicgen.py#L502

Reply:
The other models should be tested for torch script though, even if `MusicgenForConditionalGeneration` can't be

Reply:
@amyeroberts comment is exactly what my PR being a hard way 🤗

Reply:
Sorry I don't fully follow - there is only one model tested in `MusicgenTest`:
https://github.com/huggingface/transformers/blob/906afa1d5c6054a641cb6abb009cdec732a5a094/tests/models/musicgen/test_modeling_musicgen.py#L502-L503

Since there is only one model specified, there is only model that is going to be tested: `MusicgenForConditionalGeneration`. This is the single checkpoint that is omitted from the torch script tests.

Surely it's simpler to set `test_torchscript=False` for this tester, rather than override the `_create_and_check_torchscript` function?

Reply:
You are right @sanchit-gandhi . My bad. I was thinking the list of all model classes are in one single model test classes (as have been done in many test files). Thank you for pointing this out. And we can go with what you suggest.

But I want to  make a remark: splitting the different model classes like this, we won't be able to create tiny models for some model classes (and here would be `MusicgenModel` and  `MusicgenForCausalLM`).

I understand you have good reasons to do the test this way, but just want you know what the consequence it has on pipeline testing.

Thanks again.

Reply:
That's even more interesting - do you know why the other Musicgen models aren't included in `all_model_classes`?

Reply:
`MusicgenForConditionalGeneration` is just a placeholder to combine the three stages of the MusicGen model under one model class:
1. Text Encoder (one-to-one the same as T5)
2. Audio LM (new)
3. Audio VQ-VAE (one-to-one the same as EnCodec)
-> so the architecture is encoder/decoder/vq-vae decoder

The AudioLM is a decoder-only architecture, so has it's own model tester, where we test the base model and model with LM head:
https://github.com/huggingface/transformers/blob/e538189931a0615b56ed96fae9f9cb3418644802/tests/models/musicgen/test_modeling_musicgen.py#L168-L169

Feel free to read through the Colab if you want to know more about how MusicGen works: https://github.com/sanchit-gandhi/notebooks/blob/main/MusicGen.ipynb

Reply:
@amyeroberts  There are 2 model tests classes:

```python
class MusicgenDecoderTest(ModelTesterMixin, GenerationTesterMixin, PipelineTesterMixin, unittest.TestCase):
    all_model_classes = (MusicgenModel, MusicgenForCausalLM) if is_torch_available() else ()
```
and
```python
class MusicgenTest(ModelTesterMixin, GenerationTesterMixin, PipelineTesterMixin, unittest.TestCase):
    all_model_classes = (MusicgenForConditionalGeneration,) if is_torch_available() else ()
```

I believe it's because, this way, the input preparation logic would be easier.
(I guess we can have a single test class with all model classes in it - like `BartModelTest`, but might need a bit more work).

Reply:
But the (already over-complex) tiny model creation and pipeline testing logic can only take one test class. So we will miss something.

So far I think we can accept this.
(In the future, I feel no one would be super motivated to rewrite the test file structure here)